src/data/import.rs

```diff
@@ -76,10 +76,12 @@ fn process_file_chunk(
 
 fn deserialize_chunk(
     headers: &StringRecord,
+    first_index: &usize,
     records_chunk: Vec<Result<StringRecord, csv::Error>>,
     context: &Arc<SyncContext>,
 ) -> anyhow::Result<Vec<Entity>> {
     let mut entities: Vec<Entity> = Vec::with_capacity(Criteria::MAX_LIMIT);
+    let mut record_counter = first_index.clone();
     for record in records_chunk {
```

Comment:
Instead of manually counting, you could also use some functional iterator methods.

For example
```rust
for (i, record) in records_chunk.into_iter().enumerate() {
```
gives you the index (starting at zero), then you would still need to add the `first_index` onto that.
You could create another iterator to cound numbers and zip them 😅 : 
```suggestion
    let mut record_counter = first_index..; // range syntax start..end (where end isn't included). Without end it goes to infinity
    for (row, record) in record_counter.zip(records_chunk.into_iter()) {
```